bundles/org.openhab.binding.senseenergy/src/main/java/org/openhab/binding/senseenergy/internal/handler/SenseEnergyBridgeHandler.java

```diff
@@ -0,0 +1,213 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.senseenergy.internal.handler;
+
+import static org.openhab.binding.senseenergy.internal.SenseEnergyBindingConstants.HEARTBEAT_MINUTES;
+import static org.openhab.binding.senseenergy.internal.SenseEnergyBindingConstants.MONITOR_THING_TYPE;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.senseenergy.internal.api.SenseEnergyApi;
+import org.openhab.binding.senseenergy.internal.api.SenseEnergyApiException;
+import org.openhab.binding.senseenergy.internal.config.SenseEnergyBridgeConfiguration;
+import org.openhab.binding.senseenergy.internal.discovery.SenseEnergyDiscoveryService;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link SenseEnergyBridgeHandler}
+ *
+ * @author Jeff James - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class SenseEnergyBridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(SenseEnergyBridgeHandler.class);
+
+    private @Nullable SenseEnergyDiscoveryService discoveryService;
+
+    private SenseEnergyApi api;
+    private SenseEnergyBridgeConfiguration config;
+
+    protected @Nullable ScheduledFuture<?> heartbeatJob;
+
+    private Set<Long> monitorIDs = Collections.emptySet();
+
+    public SenseEnergyBridgeHandler(final Bridge thing, HttpClient httpClient) {
+        super(thing);
+        api = new SenseEnergyApi(httpClient);
+
+        config = getConfigAs(SenseEnergyBridgeConfiguration.class);
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(SenseEnergyDiscoveryService.class);
+    }
+
+    public SenseEnergyApi getApi() {
+        return api;
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(this::goOnline);
```

Comment:
I would prefer to add a check to validate the configuration (check if username or password is not blank.)

Reply:
doesn't the implementation in the BaseThingHandler::validateConfigurationParameters do this for required fields?

Reply:
Partial true, it does not check for whitespaces or valid mailaddress pattern etc. It only checks if _any_ value is set, and only for UI managed things.
File based configurations do not have this check and fully rely on this initialize method.